app/api/api_v1/endpoints/report.py

```diff
@@ -0,0 +1,143 @@
+import datetime
+import json
+
+from fastapi import APIRouter, Depends, Response, HTTPException
+from fpdf import FPDF
+from sqlalchemy.orm import Session
+
+import crud
+import utils
+from api import deps
+from models import User
+from schemas import FarmProfile, MachineryAssetsOfFarm, PlotParcelDetail, ReportDB, ReportCreate, Message, ReportDBID
+
+from utils import plant_protection, irrigations, fertilisation, harvests, work_book
+
+router = APIRouter()
+
+
+@router.get("/{report_id}")
+def get_by_id(
+        report_id: int,
+        current_user: User = Depends(deps.get_current_user),
+        db: Session = Depends(deps.get_db)
+):
+    """
+    Returns a report via ID.
+    """
+
+    report_db = crud.report.get(db=db, id=report_id)
+
+    if not report_db:
+        raise HTTPException(
+            status_code=400,
+            detail="Report with ID:{} does not exist.".format(report_id)
+        )
+
+    types = {"work_book": "work-book", "plant_protection": "plant-protection", "irrigation": "irrigations", "fertilisations": "fertilisations", "harvest": "harvests", "global_gap": "GlobalGAP"}
+
+    if report_db.type == "work-book":
+        json_file = json.loads(report_db.file)
+
+        pdf = utils.work_book(
```

Comment:
I see that you are creating the pdf reports on the fly every time they are requested. This is ok for the project now, but in the future we would like to save in the `Report` model the pdf file and when requested to just serve the file.

In that way we create the pdf on upload (offline) and the API will be a lot faster!

Reply:
Agreed, as the reports are relatively small and take a small amount of work to be created, it's fast enough. But I'll make a feature issue to change to a store the report in the database and fetch when needed method.